clippy_utils/src/lib.rs

```diff
@@ -1797,6 +1797,53 @@ pub fn is_expr_identity_function(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum VecInitKind {
```

Comment:
I like the addition of the extra fields :+1:, I think `clippy_utils::higher` would be a better place for this and the `get_vec_init_kind` function though :upside_down_face:

Reply:
Thanks for the suggestion. I also felt that `lib.rs` felt a little off for them.